chain/bitcoind_events.go

```diff
@@ -32,12 +32,12 @@ type BitcoindEvents interface {
 }
 
 // Ensure rpcclient.Client implements the rpcClient interface at compile time.
-var _ rpcClient = (*rpcclient.Client)(nil)
+var _ batchClient = (*rpcclient.Client)(nil)
 
 // NewBitcoindEventSubscriber initialises a new BitcoinEvents object impl
 // depending on the config passed.
 func NewBitcoindEventSubscriber(cfg *BitcoindConfig,
-	client *rpcclient.Client) (BitcoindEvents, error) {
+	client, bClient *rpcclient.Client) (BitcoindEvents, error) {
```

Comment:
nit: I think `client` & `bClient` should be defined with their interfaces rather than both being `*rpcclient.Client` which is a little confusing. Alternatively, `bClient batchClient` works.

Reply:
updated the method signature - yeah I also think it's a bit confusing and prone to errors, and think this needs to be changed in `btcd` so we can have a `rpcclient.BatchClient`.